### ISSUE TITLE
HexDumpBytes()'s writePosition returns stack-noise?

### ISSUE THREAD

**Issue by jonross-st** [OpenHFT/Chronicle-Bytes#132]
kotlin code:

```
val buf = HexDumpBytes()
buf.ensureCapacity(1024*16 )
buf.clear()
buf.writeSkip( 17 )
println( buf.writePosition() )

```
prints out "214748364817" ????

change the type of buf to `val buf = Bytes.allocateDirect( 1024 * 16)` and it prints out 17

here is the code for HexDumpBytes's writePosition(). I didn't take much time to understand the implemtation of this class, but at a casual glace this looks wrong-ish.

```
    @Override
    public long writePosition() {
        return base.writePosition() | (text.writePosition() << 32);
    }
```

**Comment by peter-lawrey** [comment]
I have added come documentation as to why this is.In short it has to track the write position of the binary and text buffers so it put both into `long` For the binary write position use `buf.writePosition() & 0xFFFFFFFFL` for the text use `buf.writePosition() >>> 32`

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.43](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.43), [BOM-2.20.67](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.67)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.101](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
